File: model/src/main/java/net/artux/pda/model/items/ItemType.java
package net.artux.pda.model.items;

public enum ItemType {
    ARMOR {
        @Override
        public int getTypeId() {
            return 4;
        }

        @Override
        public boolean isWearable() {
            return true;
        }

        @Override
        public boolean isCountable() {
            return false;
        }
    },
    PISTOL {
        @Override
        public int getTypeId() {
            return 0;
        }

        @Override
        public boolean isWearable() {
            return true;
        }

        @Override
        public boolean isCountable() {
            return false;
        }
    },
    RIFLE {
        @Override
        public int getTypeId() {
            return 1;
        }

        @Override
        public boolean isWearable() {
            return true;
        }

        @Override
        public boolean isCountable() {
            return false;
        }
    },
    MEDICINE {
        @Override
        public int getTypeId() {
            return 6;
        }

    },
    ARTIFACT {
        @Override
        public int getTypeId() {
            return 3;
        }

        @Override
        public boolean isWearable() {
            return true;
        }
    },
    DETECTOR {
        @Override
        public int getTypeId() {
            return 5;
        }

        @Override
        public boolean isWearable() {
            return true;
        }

        @Override
        public boolean isCountable() {
            return false;
        }
    },
    BULLET {
        @Override
        public int getTypeId() {
            return 2;
        }

    },
    ITEM {
        @Override
        public int getTypeId() {
            return 7;
        }

    };

    public abstract int getTypeId();

    public static ItemType getByTypeId(int typeId) {
        for (ItemType type : ItemType.values()) {
            if (type.getTypeId() == typeId)
                return type;
        }
        return ItemType.ITEM;
    }

    public boolean isWearable() {
        return false;
    }

    public boolean isCountable() {
        return true;
    }
}

File: model/src/main/java/net/artux/pda/model/items/ItemType.kt
package net.artux.pda.model.items

enum class ItemType {
    ARMOR {
        override val typeId: Int
            get() = 4
        override val isWearable: Boolean
            get() = true
        override val isCountable: Boolean
            get() = false
    },
    PISTOL {
        override val typeId: Int
            get() = 0

        override val isWearable: Boolean
            get() = true

        override val isCountable: Boolean
            get() = false
    },
    RIFLE {
        override val isWearable: Boolean
            get() = true
        override val isCountable: Boolean
            get() = false

        override val typeId: Int
            get() = 1
    },
    MEDICINE {
        override val typeId: Int
            get() = 6
    },
    ARTIFACT {
        override val typeId: Int
            get() = 3
        override val isWearable: Boolean
            get() = true
    },
    DETECTOR {
        override val isWearable: Boolean
            get() = true
        override val isCountable: Boolean
            get() = false
        override val typeId: Int
            get() = 5
    },
    BULLET {
        override val typeId: Int
            get() = 2
    },
    ITEM {
        override val typeId: Int
            get() = 7
    };

    abstract val typeId: Int
    open val isWearable: Boolean
        get() = false
    open val isCountable: Boolean
        get() = true

    companion object {
        fun getByTypeId(typeId: Int): ItemType {
            for (type in values()) {
                if (type.typeId == typeId) return type
            }
            return ITEM
        }
    }
}